econml/dowhy.py

```diff
@@ -0,0 +1,226 @@
+            treatment_names = [f"T{i}" for i in range(T.shape[1])]
+        if feature_names is None:
+            feature_names = [f"X{i}" for i in range(X.shape[1])]
+        if confounder_names is None:
+            confounder_names = [f"W{i}" for i in range(W.shape[1])]
+        if instrument_names is None:
+            instrument_names = [f"Z{i}" for i in range(Z.shape[1])]
+        column_names = outcome_names + treatment_names + feature_names + confounder_names + instrument_names
+        df = pd.DataFrame(np.hstack((Y, T, X, W, Z)), columns=column_names)
+        self.dowhy_ = CausalModel(
+            data=df,
+            treatment=treatment_names,
+            outcome=outcome_names,
+            graph=graph,
+            common_causes=feature_names + confounder_names if X.shape[1] > 0 or W.shape[1] > 0 else None,
+            instruments=instrument_names if Z.shape[1] > 0 else None,
+            effect_modifiers=feature_names if X.shape[1] > 0 else None,
+            estimand_type=estimand_type,
+            proceed_when_unidetifiable=proceed_when_unidentifiable,
+            missing_nodes_as_confounders=missing_nodes_as_confounders
+        )
+        self.identified_estimand_ = self.dowhy_.identify_effect(proceed_when_unidentifiable=True)
+        method_name = "backdoor." + self._cate_estimator.__module__ + "." + self._cate_estimator.__class__.__name__
+        init_params = {}
+        for p in self._get_params():
+            init_params[p] = getattr(self._cate_estimator, p)
+        self.estimate_ = self.dowhy_.estimate_effect(self.identified_estimand_,
+                                                     method_name=method_name,
+                                                     control_value=control_value,
+                                                     treatment_value=treatment_value,
+                                                     target_units=target_units,
+                                                     method_params={
+                                                         "init_params": init_params,
+                                                         "fit_params": kwargs,
+                                                     },
+                                                     )
+        return self
+
+    def refute_estimate(self, *, method_name, **kwargs):
+        """Refute an estimated causal effect.
+
+        If method_name is provided, uses the provided method. In the future, we may support automatic
+        selection of suitable refutation tests.
+        Following refutation methods are supported:
+            * Adding a randomly-generated confounder: "random_common_cause"
+            * Adding a confounder that is associated with both treatment and outcome: "add_unobserved_common_cause"
+            * Replacing the treatment with a placebo (random) variable): "placebo_treatment_refuter"
+            * Removing a random subset of the data: "data_subset_refuter"
+
```

Comment:
Can we point to the docs of the `refute_estimate` method in DoWhy for more info

Reply:
Done